Jenkinsfile

```diff
@@ -229,6 +230,152 @@ def runCypressE2e(){
+
+        inventoriesRprt = sh(returnStdout: true, script: '''curl -X GET ''' + palamida_url + '''/codeinsight/api/project/inventory/21 -H "Authorization: Bearer $jwt" ''')
+    }
+
+    writeFile(file: 'inventoriesRprt.report.json', text: inventoriesRprt)
+
+    def slurper = new JsonSlurperClassic().parseText(inventoriesRprt.toString().trim())
+
+    def vulnerabilities = []
+    slurper.inventoryItems.each { key, value ->
+        key.vulnerabilities.each { key1, value1 ->
+            if(key1.vulnerabilityCvssV3Severity == 'CRITICAL' || key1.vulnerabilityCvssV3Severity == 'HIGH'){
+                def msg = "    inventory name: $key.name; volnarability name: $key1.vulnerabilityName; with Severity V2: $key1.vulnerabilityCvssV2Severity; with Severity V3: $key1.vulnerabilityCvssV3Severity; "
+                vulnerabilities.add(msg)
+            }
+        }}
+
+    archiveArtifacts artifacts: 'inventoriesRprt.report.json'
+
+    return vulnerabilities
+
+}
+
+void UnitTest(){
+    script{
+        props = readProperties file:'data-hub/pipeline.properties';
+        copyRPM 'Release','10.0-5'
+        setUpML '$WORKSPACE/xdmp/src/Mark*.rpm'
+        sh 'export JAVA_HOME=`eval echo "$JAVA_HOME_DIR"`;export GRADLE_USER_HOME=$WORKSPACE$GRADLE_DIR;export M2_HOME=$MAVEN_HOME/bin;export PATH=$JAVA_HOME/bin:$GRADLE_USER_HOME:$PATH:$MAVEN_HOME/bin;cd $WORKSPACE/data-hub;rm -rf $GRADLE_USER_HOME/caches;set +e;./gradlew clean;./gradlew marklogic-data-hub:bootstrap -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;sleep 10s;./gradlew marklogic-data-hub-central:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/ |& tee console.log;sleep 10s;./gradlew ml-data-hub:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;./gradlew web:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;sleep 10s;./gradlew marklogic-data-hub-spark-connector:test -i --stacktrace -PnodeDistributionBaseUrl=http://node-mirror.eng.marklogic.com:8080/;'
+        junit '**/TEST-*.xml'
+        cobertura coberturaReportFile: '**/cobertura-coverage.xml'
+        jacoco()
+        def output=readFile 'data-hub/console.log'
+        def result=false;
+        if(output.contains("npm ERR!")){
+            result=true;
+        }
+        if(result){
+            currentBuild.result='UNSTABLE'
+        }
+        if(env.CHANGE_TITLE){
+            JIRA_ID=env.CHANGE_TITLE.split(':')[0]
+            jiraAddComment comment: 'Jenkins Unit Test Results For PR Available', idOrKey: JIRA_ID, site: 'JIRA'
+        }
+        if(!env.CHANGE_URL){
+            env.CHANGE_URL=" "
+        }
+    }
+}
+
+void PreBuildCheck() {
+ if(env.CHANGE_ID){
+   if(PRDraftCheck()){ sh 'exit 1' }
+   def reviewState=getReviewState()
+   if((!env.CHANGE_TITLE.startsWith("DHFPROD-")) && (!env.CHANGE_TITLE.startsWith("DEVO-"))){
```

Comment:
Should be || instead of &&

Reply:
I believe it has to be && since it negative condition: 
**!**env.CHANGE_TITLE.startsWith("DHFPROD-")